ice40/tests/picosoc/Makefile

```diff
@@ -15,3 +15,18 @@ TESTBENCH_SOURCES = \
 	spiflash.v
 
 include ../../../make/tests.mk
+
+firmware.hex: firmware.elf
+	riscv32-unknown-elf-objcopy -O verilog firmware.elf /dev/stdout | sed -e '1 s/@00000000/@00100000/; 2,65537 d;' > firmware.hex
+
+firmware.bin: firmware.elf
+	riscv32-unknown-elf-objcopy -O binary firmware.elf /dev/stdout | tail -c +1048577 > firmware.bin
+
+prog-firmware: firmware.hex
```

Comment:
s/hex/bin/

Reply:
You mean check in *both* the `.bin` and `.hex` rather than the `.elf`?

Reply:
oh right the hex is needed for simulation... Then I guess I'd vote for adding all the rules to generate them from source.